File: MultiBackend_HotelReservation/src/main/java/ma/ensa/full_backend/model/Reservation.java
package ma.ensa.full_backend.model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import jakarta.xml.bind.annotation.XmlTransient;
import jakarta.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import ma.ensa.full_backend.model.Client;
import java.util.Date;
import java.util.List;

@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "client_id", nullable = false)
    private Client client;

    @XmlTransient
    @OneToMany(mappedBy = "reservation", fetch = FetchType.EAGER)
    @JsonManagedReference
    private List<Chambre> chambres;

    @Temporal(TemporalType.DATE)
    private Date checkInDate;
    @Temporal(TemporalType.DATE)
    private Date checkOutDate;

//    @Enumerated(EnumType.STRING)
//    private TypeChambre typeChambre;

}

File: app/src/main/java/ma/ensa/projet/data/Reservation.kt
package ma.ensa.projet.data

// Updated data classes
data class Client(val id: Long)
data class Chambre(val id: Long)

data class Reservation(
    val id: Long? = null,
    val checkInDate: String,
    val checkOutDate: String,
    val client: Client,
    val chambres: List<Chambre>? = null
)

data class ReservationRequest(
    val checkInDate: String,
    val checkOutDate: String,
    val client: Client,
    val chambres: List<Chambre>
)